app/Main.hs

```diff
@@ -1,27 +1,90 @@
-{-# LANGUAGE ViewPatterns #-}
-
 module Main where
 
-import           Data.Text              (Text)
-import qualified Data.Text.IO           as TIO
-import           Formatting             (int, sformat, (%))
-import           System.Environment     (getArgs)
+import           Control.Applicative    ((<**>))
+import           Data.List              (find, intercalate)
+import           Data.Monoid            ((<>))
+import           Options.Applicative    (Parser, ParserInfo, ReadM, action, argument, command,
+                                         completeWith, eitherReader, execParser, fullDesc, help,
+                                         helper, hsubparser, info, metavar, progDesc, str)
 
 import           HaskellRobot.Converter (TexConverter, toTexCwVariant, toTexTheoryMin)
 import           HaskellRobot.Runner    (TaskContext (..), generateTexFile)
 
-chooseTexConverter :: Int -> Either Text TexConverter
-chooseTexConverter 0 = Right   toTexTheoryMin
-chooseTexConverter 1 = Right $ toTexCwVariant 1
-chooseTexConverter 2 = Right $ toTexCwVariant 2
-chooseTexConverter n = Left  $ sformat ("Not known cw mode: "%int) n
+data Command =
+    Generate GenerateCommand
+
+data GenerateCommand = GenerateCommand
+    { documentType     :: GenerationType
+    , studentsFileName :: FilePath
+    , tasksFileName    :: FilePath
+    , outputFolder     :: FilePath
+    }
+
+data GenerationType = TheoryMin | Test1 | Test2
+
+generationTypes :: [(String, GenerationType)]
+generationTypes =
+    [ ("theorymin", TheoryMin)
+    , ("test1", Test1)
+    , ("test2", Test2)
+    ]
+
+documentTypeNames :: String
+documentTypeNames = intercalate ", " documentTypeNamesList
+documentTypeNamesList :: [String]
+documentTypeNamesList = fst <$> generationTypes
+
+generationType :: ReadM GenerationType
+generationType = eitherReader $ \arg ->
```

Comment:
This parser is quite ugly. It's better to use [`flag'`](https://github.com/serokell/importify/blob/09f31d851b2152ae6ce880b81abc3554ade29f37/app/Options.hs#L86) combinator from `optparse-applicative` library to write parsers for enum-like types. You can see some example [here](https://github.com/serokell/importify/blob/09f31d851b2152ae6ce880b81abc3554ade29f37/app/Options.hs#L86).